cou/apps/base.py

```diff
@@ -371,16 +374,8 @@ def channel_codename(self) -> OpenStackRelease:
         :return: OpenStackRelease object
         :rtype: OpenStackRelease
         """
-        try:
-            # get the OpenStack release from the channel track of the application.
-            os_track_release_channel = OpenStackRelease(self._get_track_from_channel(self.channel))
-        except ValueError:
-            logger.debug(
-                "The current channel of '%s' does not exist or is unexpectedly formatted",
-                self.name,
-            )
-            os_track_release_channel = self.current_os_release
-        return os_track_release_channel
```

Comment:
I'm not understanding how this is connected to the Machine object introduction

Reply:
Let's ask the author of original [commit](https://github.com/canonical/charmed-openstack-upgrader/pull/226/commits/5ab9ca74e5886f75e56f5eb9e8dec1c1bbaf6a1c), because I'm not sure.

Reply:
Well it seems that I mixed a little bit the scope of the PR... hahaha